datadog_checks_base/datadog_checks/checks/base.py

```diff
@@ -98,6 +109,14 @@ def __init__(self, *args, **kwargs):
             ],
         }
 
+        # Setup metric limits
+        try:
+            metric_limit = self.instances[0].get("max_returned_metrics", self.DEFAULT_METRIC_LIMIT)
+        except Exception:
+            metric_limit = self.DEFAULT_METRIC_LIMIT
+        if metric_limit > 0:
```

Comment:
I wanted to handle the `0` case because `AgentCheck` does not have a limit, but it's not that hard to actually enforce a limit if set by the class.
Updated the code to forbid removing the limit if set by the class via DEFAULT_METRIC_LIMIT.